plasmapy/particles/particle_class.py

```diff
@@ -2210,12 +2220,12 @@ def symbol(self) -> str:
 
         If no symbol was defined, then return the value given by `repr`.
         """
-        return self._symbol
+        return repr(self) if self._symbol is None else self._symbol
```

Comment:
This changes also guarantees that the repr displayed when symbol is accessed  while no symbol was provided before is still valid even if the charge and mass have been modified.

Reply:
Yeah, I think this is a good idea.